opacclient/opacapp/src/main/java/de/geeksfactory/opacclient/frontend/StarredFragment.java

```diff
@@ -256,6 +290,166 @@ protected void export() {
                 getResources().getString(R.string.share)));
     }
 
+    public void exportToStorage() {
+        Intent intent = null;
+        //android 4.4+; use Storage Access Framework
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.KITKAT) {
+            intent = new Intent(Intent.ACTION_CREATE_DOCUMENT);
+            intent.addCategory(Intent.CATEGORY_OPENABLE);
+            // Create a file with the requested MIME type.
+            intent.setType("application/json");
+            intent.putExtra(Intent.EXTRA_TITLE, "share.json");
+            startActivityForResult(intent, REQUEST_CODE_EXPORT);
+        } else {        // <android 4.4; share raw json file
+            if(Environment.getExternalStorageState()==Environment.MEDIA_MOUNTED) {
+                File dest = new File(getActivity().getExternalCacheDir(),"starred_export.json");
```

Comment:
Do we really need the cached file? Couldn't we just pass the JSON content as ``Intent.EXTRA_TEXT`` instead? Intents can handle data on the order of a couple of hundreds of kilobytes which should be enough, at least for an old-device fallback solution that we don't really want to support. @johan12345 what do you think?

Reply:
I agree, we should just share the JSON text instead or not offer the feature on Android <4.4 at all. Android 4.4 and above is already [91% of all Android devices](https://developer.android.com/about/dashboards/index.html).

Reply:
It has been done!